lib/parachain/validation_protocol_test.go

```diff
@@ -4,11 +4,314 @@
 package parachain
 
 import (
+	_ "embed"
+	"fmt"
 	"testing"
 
+	"github.com/ChainSafe/gossamer/lib/common"
+	parachaintypes "github.com/ChainSafe/gossamer/lib/parachain/types"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
 )
 
+//go:embed testdata/validation_protocol.yaml
+
+var testValidationProtocolHexRaw string
+
+var testValidationProtocolHex map[string]string
+
+func init() {
+	err := yaml.Unmarshal([]byte(testValidationProtocolHexRaw), &testValidationProtocolHex)
+	if err != nil {
+		fmt.Printf("Error unmarshaling test data: %s\n", err)
+		return
+	}
+}
+
+func TestMarshalUnMarshalValidationProtocol(t *testing.T) {
+	t.Parallel()
+	/* ValidationProtocol with ApprovalDistribution with Assignments Rust code
+	fn try_msg_assignments_encode() {
```

Comment:
I don't think if it is useful to have rust code here as explanation since we already have our own test in go and it's working ok. 
Can we add the permalink from github instead putting the code?

Reply:
I'd like to keep the example rust code here so that reading can confirm how the encoding is generated.  I haven't committed the rust code back to repo, so I don't think I can create a github permalink.

Reply:
Oh my bad, I thought it was from substrate 😅